### PR TITLE
introduce the quickstart mode user guide

### PR DESCRIPTION
**Pull request by Cccccrystall**
### Purpose
Introduce the Quickstart mode user guide for the Docker deployment website.

### CODE REVIEW

**Comment by Cccccrystall** [comment]
Hi, @bossenti! I have updated the Quickstart introduction and user guide.
In the `05_deploy_docker.md`, I add a link to `user guide` at the end of Quickstart Mode Introduction. I don't split the user guide and example asset introdution, to avoid user confusion about too many links. Consequently, in the user guide, you could see the different module introdution and the example asset introdution.

File: docs/user-guide-for-quickstart.md [hunk 1]
```diff
@@ -0,0 +1,133 @@
+---
+id: user-guide-for-quickstart
+title: Quickstart Tour
+sidebar_label: Tour
+---
+
+StreamPipes is a framework that enables domain experts to model and execute stream processing pipelines in a big data infrastructure.
```
Comment (bossenti) [inline docs/user-guide-for-quickstart.md:7]: This is not our value proposition.

> Apache StreamPipes is a self-service Industrial IoT toolbox to enable non-technical users to connect, analyze and explore IoT data streams.

File: docs/user-guide-for-quickstart.md [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+The graphical user interface of StreamPipes is a web application that provides an easy to use solution for domain experts.
+
+Quickstart Mode, ideal for first-time users, allows users to interact with pre-set pipelines, dashboards, and data views to experience StreamPipes' functionalities for IIoT. 
+This page provides an overview of StreamPipes' features in Quickstart Mode, guiding users on how to utilize them effectively.
+
```
Comment (bossenti) [inline docs/user-guide-for-quickstart.md:12]:
```suggestion
:::info
Do you have StreamPipes already running in the quickstart mode? If not simply run the command `docker-compose -f docker-compose.quickstart.yml up -d` from the directory `installer/compose`.
:::
```

File: docs/user-guide-for-quickstart.md [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+On the left, the navigation menu can be seen.
+You can either use the icons on the left side or click on the menu icon on the
+top left to open the details view of the navigation menu.
+
+<div class="my-carousel docs-carousel">
+    <img src="/img/quickstart/user-guide/01_home.png" alt="Home"/>
+</div>
+
+
+## Pipelines
```
Comment (bossenti) [inline docs/user-guide-for-quickstart.md:31]: Can we please start with the Connect module? Simply switch the ordering

File: docs/user-guide-for-quickstart.md [hunk 4]
```diff
@@ -0,0 +1,133 @@
...
+</div>
+
+## Connect
+The _Connect_ view provides an overview of all existing adapters to let StreamPipes connect data sources.
+
+With StreamPipes Connect it is possible to connect new data sources in StreamPipes with just a few clicks.
+It is also possible to connect specific data sources or connect generic sources like message brokers or databases.
+If the event schema of the data source is unknown, the system tries to infer the schema by extracting some sample data and analysing it.
+
+In the Quickstart Mode, we use the Machine Data Simulator and Data stream to simulate the factory machine condition data and environmental data, just click 'START ALL ADAPTERS' button, we could start all existing adapters to offer data source for pipelines.
```
Comment (bossenti) [inline docs/user-guide-for-quickstart.md:69]:
```suggestion
In the Quickstart Mode, we use the Machine Data Simulator and Data stream to simulate the factory machine condition data and environmental data, just click the `START ALL ADAPTERS` button. This starts all existing adapters to offer data sources for pipelines.
```

File: docs/user-guide-for-quickstart.md [hunk 5]
```diff
@@ -0,0 +1,133 @@
...
+A pipeline in Apache StreamPipes describes the transformation process from a data stream to a data sink. Typically, a pipeline consists of at least one data stream (or data set), zero or more data processors and at least one data sink.
+Existing pipelines can be managed using this pipeline view.For instance, users can start and stop pipelines or delete them when they are not longer needed.
+In the pipeline page, we could see there are 5 pipelines, click 'START ALL PIPELINES' button, we could start all the predefined pipelines.
+
+<div class="my-carousel docs-carousel">
+    <img src="/img/quickstart/user-guide/02_pipelines.png" alt="Pipeline Overview"/>
+</div>
+
+
+ Using the 'Water level trend analysis' pipeline as an example. By clicking the 'Show pipeline' button, we could see the details of the pipeline as the following figure.
```
Comment (bossenti) [inline docs/user-guide-for-quickstart.md:43]: Please change order with the image

File: docs/user-guide-for-quickstart.md [hunk 6]
```diff
@@ -0,0 +1,133 @@
...
+ 
+
+The pipeline contains 3 major parts:
```
Comment (bossenti) [inline docs/user-guide-for-quickstart.md:46]: Would be great if you could provide some rationale _why_ one would like to build such a pipeline. What is the use case for it?

File: docs/user-guide-for-quickstart.md [hunk 7]
```diff
@@ -0,0 +1,133 @@
...
+    <img src="/img/quickstart/user-guide/03_adapters.png" alt="Adapters"/>
+</div>
+
+
+
+## Dashboard
+The _Dashboard_ can be used for multiple use cases.
+
+It is a good way to live monitor a running system in a dashboard, but it can also be used during the pipeline development to get a fast feedback from newly created pipelines.
+Currently 10 different types are available, including line charts, various map visualizations and data tables. The Dashboard feature allows you to select and view real-time data from specific pipelines that interest you.
```
Comment (bossenti) [inline docs/user-guide-for-quickstart.md:82]:
```suggestion
Currently 10 different types of visualizations are available, including line charts, various map visualizations and data tables. The Dashboard feature allows you to select and view real-time data from specific pipelines that interest you.
```

File: docs/user-guide-for-quickstart.md (hunk 5 above)
Comment (bossenti) [reply docs/user-guide-for-quickstart.md:43]: I was just referring to the fact that the image appears above the statement and not below

Comment (Cccccrystall) [reply docs/user-guide-for-quickstart.md:43]: Ah, I got it! Thanks for reminding me.

File: docs/05_deploy-docker.md [hunk 8]
```diff
@@ -31,6 +31,10 @@ Tested on: **macOS, Linux, Windows 10** (CMD, PowerShell, GitBash)
 We provide several options to get you going:
 
 - **default**: Default docker-compose file, called `docker-compose.yml`.
+- **nats**: The standard installation which uses Nats as message broker,called `docker-compose.nats.yml`.
+- **full**: Contains experimental Flink wrappers, called `docker-compose.full.yml`.
+- **quickstart**: Contains predefined example assets, called `docker-compose.quickstart.yml`. The Quickstart mode is a user-friendly feature which comes with predefined example assets like pipelines, dashboards, and data views. These ready-to-use components allow first-time users to get a feel of StreamPipes in IIoT with ease, serving as a practical demonstration of how StreamPipes can be utilized for efficient monitoring and analysis. We highly recommend first-time users to begin with the Quickstart mode to understand the simplicity and convenience that StreamPipes brings to the IIoT platform. A user guide is offer in the link [User Guide for Quickstart Mode](user-guide-for-quickstart.md).
```
[the remark this answers is not in the document]
Comment (Cccccrystall) [reply docs/05_deploy-docker.md]: Hi, @bossenti ! 
I have changes the technical terms in the code format, like `START ALL PIPELINES`.
In addition, in the [https://streampipes.apache.org/docs/try-installation/](url), I add the Quickstart breif introduction and a link to [https://streampipes.apache.org/docs/next/deploy-docker/](url).

File: docs/user-guide-for-quickstart.md (hunk 2 above)
Comment (Cccccrystall) [reply docs/user-guide-for-quickstart.md:11]: Hi, @bossenti ! I have added the part as the suggestion, The `info` format looks really cool! Thanks a lot!

File: docs/user-guide-for-quickstart.md (hunk 3 above)
Comment (Cccccrystall) [reply docs/user-guide-for-quickstart.md:46]: Hi, @bossenti ! I have put the `Connect` module as the first module after `Home`.

File: docs/user-guide-for-quickstart.md (hunk 4 above)
Comment (Cccccrystall) [reply docs/user-guide-for-quickstart.md]: Hi, @bossenti ! have changed the description as suggestion.

File: docs/user-guide-for-quickstart.md (hunk 7 above)
Comment (Cccccrystall) [reply docs/user-guide-for-quickstart.md]: Hi, @bossenti ! I have updated the description in this part.

File: docs/user-guide-for-quickstart.md (hunk 6 above)
Comment (Cccccrystall) [reply docs/user-guide-for-quickstart.md:66]: Hi, @bossenti ! Thanks for the suggestion!
 I have added the description of this part in 59 row as follows:
In smart factory , monitoring and analysing machine parameters are essential steps to ensure machine safety, and minimizing downtime. In the industrial machine cooling system or hydraulic machines, water level could be an important part of monitoring parameters to ensure machine's performance and safety.

File: docs/user-guide-for-quickstart.md [hunk 9]
```diff
@@ -0,0 +1,133 @@
...
+---
+
+StreamPipes is a framework that enables domain experts to model and execute stream processing pipelines in a big data infrastructure.
+The graphical user interface of StreamPipes is a web application that provides an easy to use solution for domain experts.
+
+Quickstart Mode, ideal for first-time users, allows users to interact with pre-set pipelines, dashboards, and data views to experience StreamPipes' functionalities for IIoT. 
+This page provides an overview of StreamPipes' features in Quickstart Mode, guiding users on how to utilize them effectively.
+
+
+First of all, a user needs to log in to the system. The default login credentials are ``admin@streampipes.apache.org``, password ``admin``
```
[the remark this answers is not in the document]
Comment (Cccccrystall) [reply docs/user-guide-for-quickstart.md]: Hi, @bossenti ! I have updated this part in latest commit.

File: docs/user-guide-for-quickstart.md (hunk 1 above)
Comment (Cccccrystall) [reply docs/user-guide-for-quickstart.md]: Hi, @bossenti ! I have updated the StreamPipes introduction.

**Comment by Cccccrystall** [comment]
Hi, @bossenti ! Should there be any need for enhancement in the documentation, please don't hesitate to indicate them. I'm always ready to make the necessary improvements! Thanks for your suggestion again! 😊

**Pull request merged**